geniza/footnotes/models.py

```diff
@@ -109,29 +119,41 @@ def __str__(self):
         # author, title
         # author, title (year)
         # author (year)
+        # author, "title" journal vol (year)
 
-        text = ""
+        author = ""
         if self.authorship_set.exists():
             author_lastnames = [a.creator.last_name for a in self.authorship_set.all()]
             # combine the last pair with and; combine all others with comma
             # thanks to https://stackoverflow.com/a/30084022
             if len(author_lastnames) > 1:
-                text = " and ".join(
+                author = " and ".join(
                     [", ".join(author_lastnames[:-1]), author_lastnames[-1]]
                 )
             else:
-                text = author_lastnames[0]
+                author = author_lastnames[0]
+
+        parts = []
 
         if self.title:
-            # delimit with comma if there is an author
-            if text:
-                text = ", ".join([text, self.title])
+            # if this is an article, wrap title in quotes
+            if self.source_type.type == "Article":
+                parts.append('"%s"' % self.title)
             else:
-                text = self.title
+                parts.append(self.title)
 
+        if self.journal:
+            parts.append(self.journal)
+        if self.volume:
+            parts.append(self.volume)
         if self.year:
-            text = "%s (%d)" % (text, self.year)
-        return text
+            parts.append("(%d)" % self.year)
+
+        # title, journal, etc should be joined by spaces only
+        ref = " ".join(parts)
+
+        # delimit with comma whichever values are set
+        return ", ".join([val for val in (author, ref) if val])
```

Comment:
this is a really complex `__str__()`! dunno if there's any way around it though. is the reason for trying to generate a format similar to the spreadsheet because it will help the team? i wonder if we could make it less complex by choosing a format that is more regular, if they don't absolutely need this format. unless it's just a problem of cobbling together many potentially missing pieces, which might not be resolvable.

Reply:
There's a lot of potential missing data, which does make it complicated. I'm open to revising the format in the future, but thought it would be helpful for now to stick with something similar to what the team is used to. (Also since I think we should provide something close to that version for populating the current PGP site).